manual/src/encryption.md

```diff
@@ -0,0 +1,774 @@
+
+With field level encryption, applications can encrypt fields in documents prior to transmitting data over the wire to the server. Client-side field level encryption supports workloads where applications must guarantee that unauthorized parties, including server administrators, cannot read the encrypted data.
+
+See also the MongoDB documentation on [Client Side Field Level Encryption](https://dochub.mongodb.org/core/client-side-field-level-encryption).
+
+## Dependencies
+
+To get started using client-side field level encryption in your project, you will need to install [libmongocrypt](https://github.com/mongodb/libmongocrypt), which can be compiled from source or fetched from a variety of package repositories; for more information, see the libmongocrypt [README](https://github.com/mongodb/libmongocrypt/blob/master/README.md).  If you install libmongocrypt in a location outside of the system library search path, the `MONGOCRYPT_LIB_DIR` environment variable will need to be set when compiling your project.
+
+Additionally, either `crypt_shared` or `mongocryptd` are required in order to use automatic client-side encryption.
+
+### crypt_shared
+
+The Automatic Encryption Shared Library (crypt_shared) provides the same functionality as mongocryptd, but does not require you to spawn another process to perform automatic encryption.
+
+By default, the `mongodb` crate attempts to load crypt_shared from the system and if found uses it automatically. To load crypt_shared from another location, set the `"cryptSharedLibPath"` field in `extra_options`:
+```rust,no_run
+# extern crate mongodb;
+# use mongodb::{bson::doc, Client, error::Result};
+#
+# async fn func() -> Result<()> {
+# let options = todo!();
+# let kv_namespace = todo!();
+# let kms_providers: Vec<_> = todo!();
+let client = Client::encrypted_builder(options, kv_namespace, kms_providers)?
+    .extra_options(doc! {
+        "cryptSharedLibPath": "/path/to/crypt/shared",
+    })
+    .build();
+#
+# Ok(())
+# }
+```
+If the `mongodb` crate cannot load crypt_shared it will attempt to fallback to using mongocryptd by default.  Include `"cryptSharedRequired": true` in the `extra_options` document to always use crypt_shared and fail if it could not be loaded.
+
+For detailed installation instructions see the [MongoDB documentation on Automatic Encryption Shared Library](https://www.mongodb.com/docs/manual/core/queryable-encryption/reference/shared-library).
+
+### mongocryptd
+
+The `mongocryptd` binary is required for automatic client-side encryption and is included as a component in the [MongoDB Enterprise Server package](https://dochub.mongodb.org/core/install-mongodb-enterprise). For detailed installation instructions see the [MongoDB documentation on mongocryptd](https://dochub.mongodb.org/core/client-side-field-level-encryption-mongocryptd).
```

Comment:
this sentence sounds not quite right, since mongocryptd isn't actually required if you're using crypt_shared.

Reply:
Yup, fixed.